### PR TITLE
Update scaffold generator to care about --api flag.

### PR DESCRIPTION
**Pull request by k3rni**
* don't generate view specs
* don't generate route specs for new and edit
* generate a slightly different controller spec, without redirects but
  with JSON expectations

Addresses #1672

### CODE REVIEW

**Comment by k3rni** [comment]
The CI failures seem to be of two kinds:

1. https://travis-ci.org/rspec/rspec-rails/jobs/150062031 - weird unrelated failure ?
2. https://travis-ci.org/rspec/rspec-rails/jobs/150062007 - Rubocop complaints. BTW it seems that Rubocop is only enabled for rubies below 2.2, is that intentional?

How should I proceed? Do you have suggestions on

**Comment by JonRowe** [comment]
I've bumped the broken build, you'll need to fix the rubocop failure though.

> BTW it seems that Rubocop is only enabled for rubies below 2.2, is that intentional?

It's a side effect of the version we're running.

**Comment by k3rni** [comment]
Per the linked issue, the easiest way to satisfy Rubocop here would be to just drop these methods. Is it okay, if that solution were to be accepted, to do that in this PR?

**Comment by JonRowe** [comment]
If you're sure they're not used, the build should fail if they are...

**Comment by samphippen** [comment]
@k3rni you'll need to update this for #1689 which removes assigns and assert templates from our specs. We definitely don't want those anymore. If you could do that, I can probably merge this down.

**Comment by k3rni** [comment]
Rebasing to current master and updating.

**Comment by k3rni** [comment]
I chose to add test for `response.location` returning the newly created object's url to `POST#create`. This is to match testing `redirect_to` in regular controllers.

File: lib/generators/rspec/scaffold/templates/api_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+<% end -%>
+        expect(response).to have_http_status(:ok)
+        expect(response.content_type).to eq('application/json')
+      end
+    end
+
+    context "with invalid params" do
+      it "renders a JSON response with errors for the <%= ns_file_name %>" do
+<% if RUBY_VERSION < '1.9.3' -%>
+        <%= file_name %> = <%= class_name %>.create! valid_attributes
```
Comment (samphippen) [inline lib/generators/rspec/scaffold/templates/api_controller_spec.rb]: @k3rni 

you missed moving this file name assignment up to outside of the 1.9.3 conditional, so this does not work on ruby interpreters greater than or equal to 1.9.3, can you add a spec for this case and make it pass? Thanks

**Comment by samphippen** [comment]
@k3rni sorry about the slow review here, it's been a crazy week at work. I've left you a review note, but apart from that, I think this looks great and I'll merge it when you're done 👍  Thank you so much ✨

**Comment by k3rni** [comment]
@samphippen so basically just move the assignment (line 140 above) up outside the conditional, and verify in tests, got it.

**Comment by JonRowe** [comment]
Thanks :heart:

**Pull request merged**